### PR TITLE
Group permission, step 5: Manager role cleanup

### LINKED ISSUE #317: Group permission, step 5: Manager role cleanup
The **end goal** will be to implement the group permissions as described in https://france-ioi.github.io/algorea-devdoc/design/access-rights/groups/

Step 5: Manager role cleanup 
Description: 
Remove `users.owned_group_id` and `groups_groups.role`

Subtasks:
* remove `users.owned_group_id` 
   - remove groups matching any `owned_group_id`
   - remove the root_admin_group_id setting and the group itself
   - adapt services (login callback, user delete, get dump, create group check, auth middleware)
   - remove column 
* remove  `groups_groups.role`
  - remove all `groups_groups` with a role != 'member'
  - remove column

### PR DESCRIPTION
**Pull request by zenovich**
Fixes #317

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -117,15 +117,15 @@ jobs:
       - run:
           name: Apply all migrations
           command: make db-migrate
-          no_output_timeout: 30m
```
Comment (smadbe) [inline .circleci/config.yml]: This becomes become too long, we should just upload a new version with the existing migration already applied. What we want to test is our new migrations anyway. 
(out of scope of this PR)

File: db/migrations/70_drop_column_users_owned_group_id.sql [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    JOIN `users` ON `users`.`owned_group_id` = `groups_groups`.`parent_group_id`;
+DELETE `groups_groups` FROM `groups_groups`
+    JOIN `users` ON `users`.`owned_group_id` = `groups_groups`.`child_group_id`;
+DELETE `groups_login_prefixes` FROM `groups_login_prefixes`
+    JOIN `users` ON `users`.`owned_group_id` = `groups_login_prefixes`.`group_id`;
+DELETE `groups` FROM `groups` JOIN `users` ON `users`.`owned_group_id` = `groups`.`id`;
+DELETE `groups_propagate` FROM `groups_propagate`
+    JOIN `users` ON `users`.`owned_group_id` = `groups_propagate`.`id`;
+
+# Remove UserAdmin from `groups`.`type`
```
[the remark this answers is not in the document]
Comment (zenovich) [reply db/migrations/70_drop_column_users_owned_group_id.sql]: fixed

File: .circleci/config.yml (hunk 1 above)
Comment (zenovich) [reply .circleci/config.yml]: There was a bug which made the migration so long. Now it is fixed.

File: db/migrations/70_drop_column_users_owned_group_id.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (smadbe) [reply db/migrations/70_drop_column_users_owned_group_id.sql]: Ok, probably the previous requests (L29-47) are not really needed anymore but ok it does not harm.

**Pull request merged**